src/main/java/org/truffleruby/cext/CExtNodes.java

```diff
@@ -463,6 +463,26 @@ public int bitLength(DynamicObject num) {
         }
     }
 
+    @CoreMethod(names = "rb_absint_singlebit_p", onSingleton = true, required = 1)
+    public abstract static class AbsintSinglebitPNode extends CoreMethodArrayArgumentsNode {
+
+        @Specialization
+        public int absintSinglebitP(int num) {
+            return Integer.bitCount(Math.abs(num)) == 1 ? 1 : 0;
```

Comment:
What if num is Integer.MIN_VALUE? I think it would be safer to call Integer#abs and the go in this method which can assert all values are `>=0`.

Reply:
Updated this to call `abs` first.